fingerprint2.js

```diff
@@ -822,6 +822,16 @@
       var elem = document.createElement("canvas");
       return !!(elem.getContext && elem.getContext("2d"));
     },
+    isWebGlSupported: function() {
+      if (!this.isCanvasSupported() || !!window.WebGLRenderingContext) {
```

Comment:
Is `!!window.WebGLRenderingContext` correct? It's double negation, returns true when supported.
I think this needs to test for positive, rather than negative, like if(this.isCanvas && !!window.WebGLRenderingContext) { true} else { false};